crates/rust-project-goals-cli/src/cfp.rs

```diff
@@ -0,0 +1,376 @@
+            .with_context(|| format!("Failed to write to SUMMARY.md"))?;
+        
+        println!("Updated SUMMARY.md with {} section", timeframe);
+    } else {
+        println!("Would update SUMMARY.md with {} section", timeframe);
+    }
+    
+    Ok(())
+}
+
+/// Updates the src/README.md with information about the new timeframe
+fn update_main_readme(timeframe: &str, lowercase_timeframe: &str, dry_run: bool) -> Result<()> {
+    let readme_path = "src/README.md";
+    let mut content = String::new();
+    File::open(readme_path)
+        .with_context(|| format!("Failed to open README.md"))?
+        .read_to_string(&mut content)
+        .with_context(|| format!("Failed to read README.md"))?;
+    
+    // Extract year and half from timeframe
+    let _year = &timeframe[0..4];
+    let half = &timeframe[4..].to_lowercase();
+    
+    // Determine the months based on the half
+    let (start_month, end_month) = if half == "h1" {
+        ("January", "June")
+    } else {
+        ("July", "December")
+    };
+    
+    // Create the new section to add with capitalized H
+    let capitalized_timeframe = format!("{}H{}", &timeframe[0..4], &timeframe[5..]);
+    let new_section = format!(
+        "\n## Next goal period ({})\n\n\
+         The next goal period will be {}, running from {} 1 to {} 30. \
+         We are currently in the process of assembling goals. \
+         [Click here](./{}/goals.md) to see the current list. \
+         If you'd like to propose a goal, [instructions can be found here](./how_to/propose_a_goal.md).\n",
+        capitalized_timeframe, capitalized_timeframe, start_month, end_month, lowercase_timeframe
+    );
+    
+    // Find a good place to insert the new section
+    // Look for an existing "Next goal period" section or add after the current goal period section
+    let mut new_content = content.clone();
+    
+    // First check for an existing entry for this specific timeframe
+    let this_period_pattern = Regex::new(&format!(r"## Next goal period(?:\s*\({}\))?\s*\n", regex::escape(&capitalized_timeframe))).unwrap();
+    
+    // If not found, look for any "Next goal period" section
+    let next_period_pattern = Regex::new(r"## Next goal period(?:\s*\([^\)]*\))?\s*\n").unwrap();
+    
+    // Also look for "Current goal period" to place after it if no "Next goal period" exists
+    let current_period_pattern = Regex::new(r"## Current goal period(?:\s*\([^\)]*\))?\s*\n").unwrap();
+    
+    // First try to find and replace an existing entry for this specific timeframe
+    if let Some(this_period_match) = this_period_pattern.find(&content) {
+        // Found an existing section for this specific timeframe
+        // Find the end of this section (next section or end of file)
+        if let Some(next_section_pos) = content[this_period_match.end()..].find("\n## ") {
+            let end_pos = this_period_match.start() + next_section_pos + this_period_match.end() - this_period_match.start();
```

Comment:
this also looks weird?

Reply:
Hmm, yeah, let me test out that scenario. Maybe I can ask it to generate some unit tests for this code.